Negative UncertainQuantity messes up dimensionality

Consider this:

```
In [1]: import quantities as pq

In [2]: s = pq.UncertainQuantity(3, pq.m, 2)

In [3]: s.rescale(pq.km)
Out[3]: UncertainQuantity(array(0.003), km, array(0.002))

In [4]: s2 = pq.UncertainQuantity(-3, pq.m, 2)

In [5]: s2.rescale(pq.km)
Out[5]: UncertainQuantity(array(-0.003), km, array(0.002))

In [6]: (-s2).rescale(pq.km)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
/home/bjorn/.local/lib/python3.4/site-packages/quantities/quantity.py in rescale(self, units)
    191         try:
--> 192             cf = get_conversion_factor(from_u, to_u)
    193         except AssertionError:

/home/bjorn/.local/lib/python3.4/site-packages/quantities/quantity.py in get_conversion_factor(from_u, to_u)
     52     to_u = to_u._reference
---> 53     assert from_u.dimensionality == to_u.dimensionality
     54     return from_u.magnitude / to_u.magnitude

AssertionError: 

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-6-cbe519da7162> in <module>()
----> 1 (-s2).rescale(pq.km)

/home/bjorn/.local/lib/python3.4/site-packages/quantities/uncertainquantity.py in rescale(self, units)
     72         cls = UncertainQuantity
     73         ret = super(cls, self).rescale(units).view(cls)
---> 74         ret.uncertainty = self.uncertainty.rescale(units)
     75         return ret
     76 

/home/bjorn/.local/lib/python3.4/site-packages/quantities/quantity.py in rescale(self, units)
    194             raise ValueError(
    195                 'Unable to convert between units of "%s" and "%s"'
--> 196                 %(from_u._dimensionality, to_u._dimensionality)
    197             )
    198         return Quantity(cf*self.magnitude, to_u)

ValueError: Unable to convert between units of "dimensionless" and "km"
```

--- a/quantities/tests/test_uncertainty.py
+++ b/quantities/tests/test_uncertainty.py
@@ -51,6 +51,13 @@ class TestUncertainty(TestCase):
         self.assertQuantityEqual(a*2, [2, 4]*pq.m)
         self.assertQuantityEqual((a*2).uncertainty, [0.2,0.4]*pq.m)
 
+    def test_uncertainquantity_negative(self):
+        a = UncertainQuantity([1, 2], 'm', [.1, .2])
+        self.assertQuantityEqual(-a, [-1., -2.]*pq.m)
+        self.assertQuantityEqual((-a).uncertainty, [0.1, 0.2]*pq.m)
+        self.assertQuantityEqual(-a, a*-1)
+        self.assertQuantityEqual((-a).uncertainty, (a*-1).uncertainty)
+
     def test_uncertainquantity_divide(self):
         a = UncertainQuantity([1, 2], 'm', [.1, .2])
         self.assertQuantityEqual(a/a, [1., 1.])

--- a/quantities/uncertainquantity.py
+++ b/quantities/uncertainquantity.py
@@ -132,6 +132,9 @@ class UncertainQuantity(Quantity):
     def __rmul__(self, other):
         return self.__mul__(other)
 
+    def __neg__(self):
+        return self*-1
+
     @with_doc(Quantity.__truediv__, use_header=False)
     def __truediv__(self, other):
         res = super(UncertainQuantity, self).__truediv__(other)

